cpp/mrc/src/public/coroutines/task_container.cpp

```diff
@@ -53,20 +54,26 @@ auto TaskContainer::start(Task<void>&& user_task, GarbageCollectPolicy cleanup)
 {
     m_size.fetch_add(1, std::memory_order::relaxed);
```

Comment:
Shouldnt `m_size` only be incremented once the task is launched? This will increment even when the task is queued

Reply:
m_size is the number of tasks in the container, not just the number of tasks currently executing.

Reply:
m_size is used to determine if there are any tasks left for cleanup and allow the final garbage collection coroutine to exit it's loop, so we need this to increment when a task is queued so we don't exit too early.

Reply:
Shouldnt we then have a separate variable for tracking the number of active tasks?

Reply:
I can't think of a reason to track the number of active tasks outside of making that information available for debugging purposes, and we don't have a use case for that yet. can you explain what that value would be used for?

Reply:
After discussing, I realized my mistake in thinking that `m_size` was the right value to use to determine if we can start another task immediately. Instead, we can calculate the number of currently executing tasks by `m_size - m_next_tasks.size()`.